CONTRIBUTING.md

```diff
@@ -15,47 +15,67 @@ Here are some important resources:
+
+6.  Add the repo as a remote (every time you clone the repo)
+    `git remote add yourname https://github.com/yourusername/harvey.git`
+    where "yourname" is the name of your github login.
+    `git remote -v` should look like this:
+    ```
+    $ git remote -v
+    yourname https://github.com/yourname/harvey.git (fetch)
+    yourname https://github.com/yourname/harvey.git (push)
+    origin https://github.com/Harvey-OS/harvey.git (fetch)
+    origin https://github.com/Harvey-OS/harvey.git (push)
+    ```
+
+7.  Push your branch to your forked repo like so `git push yourname feature-name`
+
+8.  Create a pull request by going to `https://github.com/yourname/harvey/pull/new/feature-name` or clicking the PR button
+    ![sevki_harvey__a_64_bit_operating_system_based_on_plan_9_from_bell_labs_and_nix__under_gpl](https://cloud.githubusercontent.com/assets/429977/13457635/79359350-e069-11e5-987b-1b4fccc45372.png)
+
+9.  Add details of what you have worked on and what your intentions are.
+    ![comparing_harvey-os_master___sevki_travis-trials_ _harvey-os_harvey](https://cloud.githubusercontent.com/assets/429977/13457683/aa2a423a-e069-11e5-84cc-1173e33264cb.png)
+    When you send a pull request, we will love you forever if you include qemu outputs and more tests. We can always use more test coverage. Please follow our coding conventions (below) and make sure all of your commits are atomic (one feature per commit).
+
+### Iterating on your work
+
+- A Github PR or a Gerrit CL roughly serve the same purpose, they are feedback loops for people to give feedback and for you to iterate on your work, so be ready to repeat steps 3, 4 and 7.
+- When iterating on your work continue committing to the same branch and push the changes up to your fork. Github will track the changes and update the PR accordingly.
+- Most of the time while you have submitted your PR, someone else will have merged something to master, at which point your branch will have been outdated, GitHub provides a convenient way of updating your branch right from your pull request.
+![build_files_by_sevki_ _pull_request__53_ _harvey-os_harvey](https://cloud.githubusercontent.com/assets/429977/13457994/4d9a3118-e06b-11e5-9898-f8574b5ce11d.png)
+- When you click that button, GitHub will update your branch, at which point you will have to `git pull yourname feature-name` and update your local repo before committing more changes.
+- If your changes are conflicting with something that was merged into the branch, you will have to resolve the changes manually and submit the changes.
+
+#### More information
+
+- [How to use Pull Requests](http://help.github.com/pull-requests/)
+- [GitHub Flow](https://guides.github.com/introduction/flow/)
+- [Hub: A command line application for GH Flow](https://hub.github.com)
+- [Video: Pull Requests](https://www.youtube.com/watch?v=81uKcXZoQ2A)
+- [Video: Workflows](https://www.youtube.com/watch?v=EwWZbyjDs9c)
 
```

Comment:
Would you mind adding some information here about rebasing and keeping your branch in sync?